packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -410,106 +570,655 @@ class _CupertinoAlertActionSection extends StatelessWidget {
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! MultiChildLayoutParentData)
+      child.parentData = new MultiChildLayoutParentData();
   }
 
   @override
-  bool shouldRepaint(_CupertinoVerticalDividerPainter other) => false;
-}
+  double computeMinIntrinsicWidth(double height) {
+    return _kCupertinoDialogWidth;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    return _kCupertinoDialogWidth;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    if (childCount == 0) {
+      return 0.0;
+    } else if (childCount == 1) {
+      // If only 1 child
+      return _computeMinIntrinsicHeightSideBySide(width);
+    } else {
+      final List<RenderBox> children = getChildrenAsList();
+
+      if (children.length == 2) {
+        if (_isSingleButtonRow(width)) {
+          // The first 2 buttons fit side-by-side. Display them horizontally.
+          return _computeMinIntrinsicHeightSideBySide(width);
+        } else {
+          // The first 2 buttons do not fit side-by-side. Display them stacked.
+          // The minimum height for 2 buttons when stacked is the minimum height
+          // of both buttons + dividers (no scrolling for 2 buttons).
+          return _computeMinIntrinsicHeightForTwoStackedButtons(width);
+        }
+      } else {
+        // 3+ buttons are always stacked. The minimum height when stacked is
+        // 1.5 buttons tall.
+        return _computeMinIntrinsicHeightStacked(width);
+      }
+    }
+  }
+
+  // The minimum height for a single row of buttons is the larger of the buttons'
+  // min intrinsic heights + the width of a divider
+  double _computeMinIntrinsicHeightSideBySide(double width) {
+    assert(childCount <= 2);
+
+    if (childCount == 1) {
+      return firstChild.computeMinIntrinsicHeight(width) + dividerThickness;
+    } else {
+      final double perButtonWidth = (width - dividerThickness) / 2.0;
+      return math.max(
+        firstChild.computeMinIntrinsicHeight(perButtonWidth) + dividerThickness,
+        lastChild.computeMinIntrinsicHeight(perButtonWidth) + dividerThickness,
+      );
+    }
+  }
+
+  // The minimum height for 2 stacked buttons is the height of both
+  // buttons + 2 dividers.
+  double _computeMinIntrinsicHeightForTwoStackedButtons(double width) {
+    assert(childCount == 2);
+
+    return (2 * dividerThickness)
+        + firstChild.computeMinIntrinsicHeight(width)
+        + lastChild.computeMinIntrinsicHeight(width);
+  }
+
+  // The minimum height for 3+ stacked buttons is the height of the 1st button
+  // + 50% the height of the 2nd button + 2 dividers.
+  double _computeMinIntrinsicHeightStacked(double width) {
+    assert(childCount >= 3);
+
+    final List<RenderBox> children = getChildrenAsList();
+    return (2 * dividerThickness)
+        + children[0].computeMinIntrinsicHeight(width)
+        + (0.5 * children[1].computeMinIntrinsicHeight(width));
```

Comment:
this is another example where getChildrenAsList is absurdly expensive for what you're doing